site/components/pages/Pledge/lib/verifyGnosisSafeMultisig.ts

```diff
@@ -0,0 +1,32 @@
+import { ethers } from "ethers";
+import { getJsonRpcProvider } from "@klimadao/lib/utils";
+import GnosisSafeSignMessageLib from "@klimadao/lib/abi/GnosisSafeSignMessageLib.json";
+
+const ONE_HOUR = 60 * 60 * 1000;
+
+export const verifyGnosisSafeMultisig = async (params: {
+  signature: string;
+  address: string;
+}): Promise<void> => {
+  const provider = getJsonRpcProvider();
+  const gnosisSafeContract = new ethers.Contract(
+    params.address,
+    GnosisSafeSignMessageLib.abi,
+    provider
+  );
+  const messageHash = ethers.utils.hashMessage(params.signature);
+  const getMessageHash = await gnosisSafeContract.getMessageHash(messageHash);
+  const signedEvent = gnosisSafeContract.filters.SignMsg(getMessageHash);
+
+  const waitForSignedEvent = (): Promise<void> =>
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        gnosisSafeContract.removeListener(signedEvent, () => resolve());
+        reject(new Error("Gnosis safe signature verification timed out"));
+      }, ONE_HOUR);
```

Comment:
Since this is waiting for signatures server-side, the lambda should time out at 60 seconds

![image](https://user-images.githubusercontent.com/88635679/191394625-8228fae8-e8df-47e2-b9af-2acf27d1098a.png)